elastic/translate.go

```diff
@@ -1,185 +1,186 @@
 package elastic
 
 import (
-	"encoding/json"
+	"errors"
 	"fmt"
-	"log"
 	"strings"
 )
 
 // Leaf is a terminal node of a query
 // All terminal query terms implement the Translate() method
-type Leaf interface {
-	Translate() map[string]interface{}
+type leaf interface {
+	Translate() Term
 }
 
-func encodeParam(concept string, param map[string]interface{}) map[string]interface{} {
-	operator := param["operator"].(string)
-	var conceptID string
-	if operator == "set" {
-		conceptID = "_id"
+// encodeParam encodes a parameter from a query to its corresponding leaf type
+func encodeParam(concept string, param map[string]interface{}) (map[string]interface{}, error) {
+	o := param["operator"].(string)
+	var cid string
+	if o == "set" {
+		cid = "_id"
 	} else {
-		conceptID = param["id"].(string)
+		cid = param["id"].(string)
 	}
 	var conceptPath string
-	if concept != "" && operator != "set" {
-		conceptPath = concept + "." + conceptID
+	if concept != "" && o != "set" {
+		conceptPath = concept + "." + cid
 	} else {
-		conceptPath = conceptID
+		conceptPath = cid
 	}
-	value := param["value"]
-	var paramLeaf Leaf
-	switch operator {
+	v := param["value"]
+	var l leaf
+	switch o {
 	case "set":
-		paramLeaf = &MemberTerm{conceptPath, value}
+		l = &MemberTerm{conceptPath, v}
```

Comment:
I hope this does not bite you later, but you are relying on the position of the fields defined in the struct for the arguments. As supposed to

```go
l = &MemberTerm{
    field: conceptPath,
    value: v,    
}
```

Generally this way is preferred, but this _is_ private code and small in scope so it is not a big deal.